keras/backend/tensorflow_backend.py

```diff
@@ -432,6 +436,10 @@ def switch(condition, then_expression, else_expression):
     return tf.python.control_flow_ops.cond(condition,
                                            lambda: then_expression,
                                            lambda: else_expression)
+def broadcasting_select(condition, A, B):
```

Comment:
> I found this perhaps more readable because otherwise the tile call is pretty inscrutable.

I think it's fine as long as there is a sufficiently detailed comment explaining what is going on (such as your comment above).